### PR TITLE
feat: continue implementing oprf-accountant v2

### PR DESCRIPTION
**Pull request by fabian1409**
> [!NOTE]
> **Medium Risk**
> Touches billing semantics, private-key signing, and a breaking DB shape; votes are not fully submitted on-chain yet, so behavior in production depends on finishing submission and aligning with the pending `currentVoteEpoch` contract API.
> 
> **Overview**
> Adds a background **OprfAccountantService** that periodically calls `BillingContract.currentVoteEpoch()`, waits for the configured post-epoch offset, aggregates per-RP request counts from Postgres for the epoch’s `[epochStart, epochEnd)` span, and builds EIP-712–signed `SignedVoteChunk`s for `submitBillingVotes` (on-chain submission is still commented out pending wallet/signer split and contract ABI).
> 
> **Storage and ingestion:** The `rp_signatures` migration is reshaped for v2—drops per-row `epoch` and the `(rp_id, nonce, epoch)` uniqueness rule, adds `action` and optional `wip101_data`, and makes `signature` nullable. Batch inserts persist the new fields; `POST /req` now returns 500 on DB errors. **`rp_counts_for_epoch_span`** supplies vote data with `COUNT(DISTINCT nonce)` per `rp_id`, ordered for the contract.
> 
> **Wiring:** Config gains billing contract address, RPC, wallet key, tick/vote timeouts, voting-window offset, and vote chunk size. Startup spawns the accountant loop alongside Axum and joins it on shutdown. Unit/integration tests cover vote-window logic, chunk signing, and aggregation edge cases. `Cargo.lock` picks up `alloy`/`web3` and `taceo-nodes-common` test-deps (e.g. `axum-test`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 90766b9fb6262fa734ec00424d573024d368cc73. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: services/oprf-accountant/src/accountant_service.rs [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+        }
+        if vote_epoch as i64 > epoch_cursor {
+            // We have the approx. voting window to submit the vote, so we can set a timeout for the entire voting window duration.
+            tokio::time::timeout(
+                voting_window_length,
+                self.vote_for_epoch(vote_epoch, counts),
+            )
+            .await??;
+            tracing::info!(vote_epoch, "successfully voted for epoch");
+            self.db.set_epoch_cursor(vote_epoch).await?;
```
[the remark this answers is not in the document]
Comment (fabian1409) [reply services/oprf-accountant/src/accountant_service.rs]: We can't vote for older epochs than current - 1 so this is fine

[the remark this answers is not in the document]
Comment (fabian1409) [reply services/oprf-accountant/src/accountant_service.rs]: Will be done in another PR

File: services/oprf-accountant/src/api.rs [hunk 2]
```diff
@@ -57,14 +59,23 @@ impl From<&NullifierOprfRequestAuthV1> for BillableRpRequest {
...
+#[instrument(level = "info", skip_all)]
 async fn post_request(
     State(db): State<PostgresDb>,
     Query(PostRequestQuery { id: _ }): Query<PostRequestQuery>,
     Json(rp_requests): Json<Vec<BillableRpRequest>>,
 ) -> impl IntoResponse {
-    // TODO get the epochs for the RpVote from the contract
-    let _ = db.store_request_batch(rp_requests).await;
-    StatusCode::OK
+    // TODO check if requests are Uniqueness Proofs (first byte is 0x00)?
```
Comment (0xThemis) [inline services/oprf-accountant/src/api.rs]: Dont necessarily think that we need to do this as we only get requests from the nodes anyways and they check it but open for discussion

File: services/oprf-accountant/src/api.rs [hunk 3]
```diff
@@ -57,14 +59,23 @@ impl From<&NullifierOprfRequestAuthV1> for BillableRpRequest {
 }
 
 // TODO add id to span
+#[instrument(level = "info", skip_all)]
```
Comment (0xThemis) [inline services/oprf-accountant/src/api.rs]: add to span

File: services/oprf-accountant/src/main.rs [hunk 4]
```diff
@@ -69,22 +79,61 @@ async fn run(config: FullOprfAccountantConfig) -> eyre::Result<()> {
...
-    tracing::info!("starting axum server on {bind_addr}",);
-    let listener = tokio::net::TcpListener::bind(bind_addr).await?;
-    axum::serve(listener, router)
-        .with_graceful_shutdown(async move { cancellation_token.cancelled().await })
-        .await
-        .context("while serving axum")?;
-    tracing::info!("axum server shutdown");
-    Ok(())
+
+    let server = tokio::spawn({
```
Comment (0xThemis) [inline services/oprf-accountant/src/main.rs]: update to not spawn axum in task

File: services/oprf-accountant/src/accountant_service.rs [hunk 5]
```diff
@@ -0,0 +1,766 @@
...
+    /// Votes for the next vote-able epoch, if any (see [`Self::maybe_get_votes`]); a no-op if no
+    /// epoch is currently vote-able.
+    ///
+    /// # Errors
+    /// Returns an error if querying the `BillingContract` or the database fails, or if voting
+    /// doesn't complete within [`OprfAccountantServiceArgs::vote_timeout`].
+    #[instrument(level = "info", skip_all, name = "accountant_service::tick")]
+    pub(crate) async fn tick(&self) -> eyre::Result<()> {
+        if let MaybeVotes::Votes { epoch, counts } = self.maybe_get_votes().await? {
+            tokio::time::timeout(self.vote_timeout, self.vote_for_epoch(epoch, counts)).await??;
```
Comment (0xThemis) [inline services/oprf-accountant/src/accountant_service.rs:350]: nit: maybe add some .context() here to the three ?. Helps in case this actually happens

**Review by 0xThemis — APPROVED** [review]

**Comment by fabian1409** [comment]
approach to billing will be different, so this is obsolete
